src/index.ts

```diff
@@ -1180,21 +1201,18 @@ export class KycDao extends ApiBase {
             if (enabledNetworkDetails.blockchain === 'Ethereum' && enabledNetworkDetails.chainId) {
               const newChainId = enabledNetworkDetails.chainId;
               try {
-                await this.evmProvider.switchNetwork(newChainId);
-
-                const updatedChainId = await this.evmProvider.getChainId();
-                if (updatedChainId !== newChainId) {
-                  throw new InternalError(
-                    `Automatic EVM network switching failed: switchNetwork call did not throw an error, but wallet returns wrong chainId (${updatedChainId}), expected ${newChainId}.`,
+                await this.switchNetworkChecked(newChainId);
+              } catch (e) {
+                if (this.isUserRejectError(e)) {
+                  throw e;
+                } else {
+                  console.log('Automatic EVM network switching failed:', e);
+                  throw new StatusError(
```

Comment:
Maybe `switchNetworkChecked` could wrap `switchNetwork` and handle this error as well, with a more generic error message so we don't have to copy this part either because of different messages. We could drop the try-catch here. Not critical.